ext/src/zpages/tracez_processor.cc

```diff
@@ -0,0 +1,37 @@
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+  void TracezSpanProcessor::OnStart(opentelemetry::sdk::trace::Recordable &span) noexcept {
+    running_spans_.insert(&span);
+  }
+
+  void TracezSpanProcessor::OnEnd(std::unique_ptr<opentelemetry::sdk::trace::Recordable> &&span) noexcept {
+     nostd::span<std::unique_ptr<opentelemetry::sdk::trace::Recordable>> batch(&span, 1);
+     if (exporter_->Export(batch) == opentelemetry::sdk::trace::ExportResult::kFailure) {
+       std::cerr << "Error batching span\n";
+     }
+
+     auto completed_span = running_spans_.find(span.get());
+     if (completed_span != running_spans_.end()) {
+       running_spans_.erase(completed_span);
+       completed_spans_.push_back(std::move(span));
+     }
+  }
+
```

Comment:
One option is to collapse these two functions into one, and return some struct that bundles copies of the running and completed spans together.

```
struct RunningAndCompletedSpans {
  <> running_spans_;
  <> completed_spans_;
};

RunningAndCompletedSpans GetRunningAndCompletedSpansSnapshot() noexcept {
  auto completed_spans = std::move(completed_spans_);
  completed_spans.clear();
  return {std::unordered_set<Recordable*>(running_spans_), completed_spans};
}
```

Something like that, that way it'll just give the Aggregator a consistent view while it's aggregating the spans.

Reply:
Okay, that sounds good! I should replace the running and completed getters in favor of returning them all, since the main goal is giving this information in a consistent manner and the aggregator should change its methods accordingly?